pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConnectionPool.java

```diff
@@ -103,20 +104,25 @@ private static class Key {
     }
 
     public ConnectionPool(InstrumentProvider instrumentProvider,
-                          ClientConfigurationData conf, EventLoopGroup eventLoopGroup) throws PulsarClientException {
-        this(instrumentProvider, conf, eventLoopGroup, () -> new ClientCnx(instrumentProvider, conf, eventLoopGroup));
+                          ClientConfigurationData conf, EventLoopGroup eventLoopGroup,
+                          ScheduledExecutorProvider scheduledExecutorProvider) throws PulsarClientException {
```

Comment:
It seems that this could simple take a ScheduledExecutorService. The coupling to ScheduledExecutorProvider seems unnecessary.